### DISCUSSION TITLE
company_id - dove trovarlo

### DISCUSSION THREAD

**Discussion by iarin-fabbri** [fattureincloud/api#263] [Q&A] [answered]
Ciao a tutti, sono nuovo di Fatture in Cloud

Stò sviluppando uno script di import/export e mi serve il company_id, ma non riesco a trovarlo da nessuna parte, magari mi sfugge qualche cosa.
Dove posso trovarlo?

**Comment by emajo** [comment, marked as answer]
Hi @iarin-fabbri, [here](https://developers.fattureincloud.it/docs/basics/company-scoped-methods/#-where-can-i-find-my-company_id) you can find what you need, I suggest to use the search feature of this discussions and the developer hub before opening new discussions.

Thanks,
Emanuele.

**Reply by iarin-fabbri** [reply]
Sorry, but I use the python SDK and cannot access directly to API. Every method use the company_id as param, but I cannot find anything in the decumentation or I cannot saw it

**Reply by valmoz** [reply]
Ciao @iarin-fabbri
i metodi esposti dalle nostre API sono di due categorie:
- Company-scoped methods -> richiedono l'utilizzo del Company ID (sono la maggioranza)
- Person scoped methods -> gli altri metodi che non richiedono il Company ID.

Come indicato nella [guida](https://developers.fattureincloud.it/docs/basics/company-scoped-methods/#-where-can-i-find-my-company_id) suggerita dal collega @emajo, lei deve usare il metodo [List User Companies](https://developers.fattureincloud.it/api-reference/#get-/user/companies), che ovviamente non richiede Company ID, per recuperare la lista delle companies della sua utenza con i relativi ID.

Solitamente questa lista avrà una sola company, quindi dovrebbe essere semplice recuperare l'ID dalla lista. Inoltre l'ID non cambia, per questo può tranquillamente effettuare una sola volta questa chiamata e salvare l'ID recuperato da qualche parte.

Per quanto riguarda la questione dell'SDK, questa libreria di fatto è solo un wrapper dell'API, quindi lei sta attivamente usando le API. Per la libreria Python il metodo è presente qui: [Python SDK - List User Companies](https://github.com/fattureincloud/fattureincloud-python-sdk/blob/master/docs/UserApi.md#list_user_companies)
Le faccio però notare che, poiché l'ID è permanente, può tranquillamente evitare di effettuare la chiamata con l'SDK e utilizzare l'[API Reference](https://developers.fattureincloud.it/api-reference#get-/user/companies) oppure [Postman](https://developers.fattureincloud.it/docs/quickstarts/postman-collection).
Tutte queste risorse (SDKs, API Reference, Postman) sono generate automaticamente a partire dallo stesso file OpenAPI, quindi contengono gli stessi metodi.

Le faccio infine notare che, per semplificare il suo lavoro, sia la community GitHub sia la nostra documentazione sono dotate di funzionalità di ricerca, che possono aiutarla nella ricerca di ciò che le serve, e inoltre che le [FAQ](https://developers.fattureincloud.it/docs/FAQs) rispondono alle domande più comuni, tra cui anche quella di cui stiamo parlando adesso.

Ci faccia sapere se è riuscito a risolvere questo problema.

Grazie
Mauro

**Reply by valmoz** [reply]
Buongiorno @iarin-fabbri è riuscito a recuperare il company_id?

Grazie
Mauro

**Reply by iarin-fabbri** [reply]
Ciao @valmoz , sì sono riuscito.
Ora ho problemi nella creazione della fattura perchè probabilmente passo dati non corretti, ma ho appena iniziato. Se non riesco a risolvere scriverò per aiuti.

Grazie per la collaborazione
